lib/command.js

```diff
@@ -223,6 +223,24 @@ module.exports = function command (yargs, usage, validation, globalMiddleware) {
     if (!yargs._hasOutput()) {
       positionalMap = populatePositionals(commandHandler, innerArgv, currentContext, yargs)
     }
+    let preCheckMiddlewares = globalMiddleware.slice(0).concat(commandHandler.middlewares || []).filter((x) => x.applyBeforeValidation === true)
+    if (preCheckMiddlewares.length > 0 && !yargs._hasOutput()) {
+      let errors = false
+      const middlewareArgs = preCheckMiddlewares.reduce(function (initialObj, middleware) {
+        let preChecksContext = {
+          'commands': JSON.parse(JSON.stringify(currentContext.commands)),
```

Comment:
I don't understand why you'd need to run `JSON.parse(JSON.stringify)` but, let's save this for a conversation related to the `context` feature 👍

Reply:
It's purely so that when passed into the middleware function the user cannot modify it and potentially cause some unintended behaviors.